tests/functional/consensus/source/p2p.ts

```diff
@@ -15,9 +47,75 @@ export class P2PRegistry {
 		return [...this.#nodes.entries()].filter(([nodeId]) => nodeId !== id).map(([, node]) => node);
 	}
 
+	public getAllNodes(): Contracts.Kernel.Application[] {
+		return [...this.#nodes.entries()].map(([, node]) => node);
+	}
+
+	getNodes(nodes?: number[]): Contracts.Kernel.Application[] {
+		if (nodes === undefined) {
+			return this.getAllNodes();
+		}
+
+		return nodes.map((node) => this.#nodes.get(node)).filter((node) => node !== undefined);
+	}
+
 	public makeBroadcaster(id: number): Broadcaster {
 		return new Broadcaster(id, this);
 	}
+
+	async postProposal(node: Contracts.Kernel.Application, proposal: Contracts.Crypto.Proposal): Promise<void> {
+		this.proposals.set(proposal);
+
+		setTimeout(async () => {
+			await node
```

Comment:
This returns a `Contracts.Consensus.ProcessorResult` - can we somehow make use of it and assert it in the tests?

E.g. that sending a invalid proposal correctly results in `Contracts.Consensus.ProcessorResult.Invalid`

(same for `postPrecommit`, `postPrevote`, `postCommit`)